src/main/kotlin/com/exactpro/th2/check1/rule/RuleFactory.kt

```diff
@@ -157,11 +159,39 @@ class RuleFactory(
                 }
             }
 
+    fun createSilenceCheck(
+        request: CheckSequenceRuleRequest,
+        timeout: Long
+    ): SilenceCheckTask {
+        return ruleCreation(request.parentEventId) {
+            checkAndCreateRule {
+                check(timeout > 0) { "timeout must be greater that zero" }
+                val sessionAlias: String = request.connectivityId.sessionAlias
+                val sessionKey = SessionKey(sessionAlias, directionOrDefault(request.direction))
+                SilenceCheckTask(
+                    request.preFilter,
+                    request.description.takeIf(String::isNotEmpty),
+                    createTaskTimeout(timeout),
+                    maxEventBatchContentSize,
+                    Instant.now(),
+                    sessionKey,
+                    request.parentEventId,
+                    streamObservable,
+                    eventBatchRouter
+                )
+            }
+            onErrorEvent {
+                Event.start()
+                    .name("Auto silence check rule cannot be created")
+                    .type("checkRuleCreation")
```

Comment:
I guess it should be a different type

Reply:
The type groups the events that care the similar information. In our case, the information that rule was not created. So, I think we should leave the type as is (it is the same for all other rules)